src/hints/mod.rs

```diff
@@ -33,96 +36,53 @@ use crate::io::InternalTransaction;
-
-    let segments_add_hint = HintFunc(Box::new(segments_add));
-    hint_processor.add_hint(String::from(hints_raw::SEGMENTS_ADD), Rc::new(segments_add_hint));
+type HintImpl = fn(
+    &mut VirtualMachine,
+    &mut ExecutionScopes,
+    &HashMap<String, HintReference>,
+    &ApTracking,
+    &HashMap<String, Felt252>,
+) -> Result<(), HintError>;
 
-    let segments_add_temp_hint = HintFunc(Box::new(segments_add_temp));
-    hint_processor.add_hint(String::from(hints_raw::SEGMENTS_ADD_TEMP), Rc::new(segments_add_temp_hint));
-
-    let transactions_len_hint = HintFunc(Box::new(transactions_len));
-    hint_processor.add_hint(String::from(hints_raw::TRANSACTIONS_LEN), Rc::new(transactions_len_hint));
-
-    let enter_syscall_scopes_hint = HintFunc(Box::new(enter_syscall_scopes));
-    hint_processor.add_hint(String::from(hints_raw::ENTER_SYSCALL_SCOPES), Rc::new(enter_syscall_scopes_hint));
-
-    let load_next_tx_hint = HintFunc(Box::new(load_next_tx));
-    hint_processor.add_hint(String::from(LOAD_NEXT_TX), Rc::new(load_next_tx_hint));
-
-    let prepare_constructor_execution_hint = HintFunc(Box::new(prepare_constructor_execution));
-    hint_processor.add_hint(String::from(PREPARE_CONSTRUCTOR_EXECUTION), Rc::new(prepare_constructor_execution_hint));
-
-    let transaction_version_hint = HintFunc(Box::new(transaction_version));
-    hint_processor.add_hint(String::from(TRANSACTION_VERSION), Rc::new(transaction_version_hint));
-
-    let assert_transaction_hash_hint = HintFunc(Box::new(assert_transaction_hash));
-    hint_processor.add_hint(String::from(ASSERT_TRANSACTION_HASH), Rc::new(assert_transaction_hash_hint));
-
-    let get_block_mapping_hint = HintFunc(Box::new(get_block_mapping));
-    hint_processor.add_hint(String::from(GET_BLOCK_MAPPING), Rc::new(get_block_mapping_hint));
-
-    let start_execute_deploy_transaction_hint = HintFunc(Box::new(start_execute_deploy_transaction));
-    hint_processor.add_hint(String::from(START_DEPLOY_TX), Rc::new(start_execute_deploy_transaction_hint));
-
-    let get_state_entry_hint = HintFunc(Box::new(get_state_entry));
-    hint_processor.add_hint(String::from(GET_STATE_ENTRY), Rc::new(get_state_entry_hint));
-
-    let check_is_deprecated_hint = HintFunc(Box::new(check_is_deprecated));
-    hint_processor.add_hint(String::from(CHECK_IS_DEPRECATED), Rc::new(check_is_deprecated_hint));
-
-    let is_deprecated_hint = HintFunc(Box::new(is_deprecated));
-    hint_processor.add_hint(String::from(IS_DEPRECATED), Rc::new(is_deprecated_hint));
-
-    let os_context_segments_hint = HintFunc(Box::new(os_context_segments));
-    hint_processor.add_hint(String::from(OS_CONTEXT_SEGMENTS), Rc::new(os_context_segments_hint));
-
-    let selected_builtins_hint = HintFunc(Box::new(selected_builtins));
-    hint_processor.add_hint(String::from(SELECTED_BUILTINS), Rc::new(selected_builtins_hint));
+pub fn sn_hint_processor() -> BuiltinHintProcessor {
```

Comment:
#54

Will re-init how we set this handler up, so maybe we should hold off on this portion until that is merged